### PR TITLE
Sorting

### PR DESCRIPTION
**Pull request by ronfaygler**

### CODE REVIEW

**Comment by ronfaygler** [comment]
@irisrifold@gmail.com

File: client/src/components/Attractions/FilterAttractions/FilterAttractions.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import styles from './FilterAttractions.module.css'; // Adjust the path if neede
 import FilterType from './FilterType';
 import {fetchFilterValues} from '../../../services/utils';
 import { AttractionContext } from '../../../context/AttractionContext';
+import Button from "@mui/material/Button";
 
 const FilterAttractions = () => {
     const {filterValuesAttractions, setFilterValuesAttractions} = useContext(AttractionContext);
```
Comment (irisri) [inline client/src/components/Attractions/FilterAttractions/FilterAttractions.js:9]: check if the data structure doesn't get mixed up with the starting state of the context (if there is in the context something like country and you want it to be city)

File: client/src/components/Attractions/ShowAttractions/ShowAttractions.js [hunk 2]
```diff
@@ -8,13 +8,13 @@ import Stack from '@mui/material/Stack';
 
 // present the attraction in home page, according to the filters
 const ShowAttractions = () => {
-  const {filterValuesAttractions, getAttractions } = useContext(AttractionContext);
+  const { filterValuesAttractions, getAttractions, chosenSort } = useContext(AttractionContext);
```
Comment (irisri) [inline client/src/components/Attractions/ShowAttractions/ShowAttractions.js:11]: weired name chosenSort what is it supposed to be?

File: client/src/components/Attractions/ShowAttractions/ShowAttractions.js [hunk 3]
```diff
@@ -8,13 +8,13 @@ import Stack from '@mui/material/Stack';
...
   const [loading, setLoading] = useState(true);
   
   // pagination: 
   let currAttractions = getAttractions();
   const [currentPage, setCurrentPage] = useState(1);
-  const [recordsPerPage] = useState(1);
+  const [recordsPerPage] = useState(4);
```
Comment (irisri) [inline client/src/components/Attractions/ShowAttractions/ShowAttractions.js:17]: if you don't have setSatate  don't use state, make it a const or useRef

File: client/src/components/Attractions/ShowAttractions/ShowAttractions.module.css [hunk 4]
```diff
@@ -49,12 +50,10 @@
 .img {
   width: 180px;
   height: 170px;
-  border-radius: 10px;
   margin-top: 10px;
-  width: 180px;
-  height: 170px;
+  /* width: 180px;
```
Comment (irisri) [inline client/src/components/Attractions/ShowAttractions/ShowAttractions.module.css:54]: delete comment

File: client/src/components/User/LoginForm.js [hunk 5]
```diff
@@ -21,7 +21,7 @@ const LoginForm = () => {
       setError(json.error);
     } else {
       const name = json.userName;
-      const profilePic = json.profilePic;
+      // const profilePic = json.profilePic;
```
Comment (irisri) [inline client/src/components/User/LoginForm.js:24]: delete comment

File: client/src/context/AttractionContext.js [hunk 6]
```diff
@@ -33,12 +34,42 @@ const AttractionProvider = ({ children }) => {
                 );
             }
         });
+        // sorting
+        if (chosenSort === 'title') {
```
Comment (irisri) [inline client/src/context/AttractionContext.js:38]: you can use function chaining, after filter use sort immediately

File: client/src/components/Attractions/ShowAttractions/ShowAttractions.js (hunk 2 above)
Comment (ronfaygler) [reply client/src/components/Attractions/ShowAttractions/ShowAttractions.js:11]: changed to selectedSortingOption

**Pull request merged**